### PR TITLE
feat: Make send return a PartialInteractionMessage

### PR DESCRIPTION
**Pull request by DenverCoder1**
## Summary

Makes a return value for `InteractionResponse.send_message` supporting `edit`, `delete` and `fetch`

## Testing

Current way of editing, deleting, fetching:

```py
@client.slash_command(guild_ids=[TEST_GUILD_ID])
async def edit_delete_old(interaction: Interaction):
    await interaction.response.send_message("Hello World")
    await interaction.edit_original_message(content=f"Message Edited")
    await interaction.delete_original_message()

@client.slash_command(guild_ids=[TEST_GUILD_ID])
async def fetch_from_interaction(interaction: Interaction):
    await interaction.response.send_message("Hello World")
    msg = await interaction.original_message()
    await msg.add_reaction("\N{THUMBS UP SIGN}")
```

New methods with `PartialInteractionMessage`:

```py
@client.slash_command(guild_ids=[TEST_GUILD_ID])
async def edit_delete_partial(interaction: Interaction):
    pm = await interaction.response.send_message("Hello World")
    await pm.edit(content=f"{pm}")
    await pm.delete()

@client.slash_command(guild_ids=[TEST_GUILD_ID])
async def fetch_from_partial(interaction: Interaction):
    pm = await interaction.response.send_message("Hello World")
    msg = await pm.fetch()
    await msg.add_reaction("\N{THUMBS UP SIGN}")
```

Full example code: https://paste.nextcord.dev/?id=1649215125063750

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -3696,6 +3696,19 @@ InteractionMessage
 .. autoclass:: InteractionMessage()
     :members:
 
+    .. automethod:: InteractionMessage.delete
+
+    .. automethod:: InteractionMessage.edit
```
Comment (Skelmis) [inline docs/api.rst:3701]: Should fetch also be documented?

Given it does exist, although it'd fetch this object so not sure.

File: nextcord/interactions.py [hunk 2]
```diff
@@ -1113,3 +1114,79 @@ async def delete(self, *, delay: Optional[float] = None) -> None:
...
+        """
+        return self._state._interaction.channel
+
+    @property
+    def guild(self) -> Optional[Guild]:
+        """Optional[:class:`Guild`]: The guild the interaction was sent from."""
+        return self._state._interaction.guild
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} author={self.author!r} channel={self.channel!r} guild={self.guild!r}>"
```
Comment (Skelmis) [inline nextcord/interactions.py:1178]: Should there be commas between args maybe?

**Review by Skelmis — APPROVED** [review]
Lgtm, just a couple comments feel free to ignore.

File: docs/api.rst (hunk 1 above)
Comment (DenverCoder1) [reply docs/api.rst:3701]: That is for InteractionMessage, not PartialInteractionMessage.

For PartialInteractionMessage, everything is documented

File: nextcord/interactions.py (hunk 2 above)
Comment (DenverCoder1) [reply nextcord/interactions.py:1178]: It seems like the other objects in Nextcord do not have commas, so I think this is the way it is supposed to be

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**